power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/newton_raphson_se_solver.hpp

```diff
@@ -21,17 +22,443 @@ namespace power_grid_model {
 // hide implementation in inside namespace
 namespace math_model_impl {
 
+// block class for the unknown vector and/or right-hand side in state estimation equation
+template <bool sym> struct NRSEUnknown : public Block<double, sym, false, 4> {
+    template <int r, int c> using GetterType = typename Block<double, sym, false, 4>::template GetterType<r, c>;
+
+    // eigen expression
+    using Block<double, sym, false, 4>::Block;
+    using Block<double, sym, false, 4>::operator=;
+
+    GetterType<0, 0> theta() { return this->template get_val<0, 0>(); }
+    GetterType<1, 0> v() { return this->template get_val<1, 0>(); }
+    GetterType<2, 0> phi_theta() { return this->template get_val<2, 0>(); }
+    GetterType<3, 0> phi_v() { return this->template get_val<3, 0>(); }
+
+    GetterType<0, 0> eta_theta() { return this->template get_val<0, 0>(); }
+    GetterType<1, 0> eta_v() { return this->template get_val<1, 0>(); }
+    GetterType<2, 0> tau_theta() { return this->template get_val<2, 0>(); }
+    GetterType<3, 0> tau_v() { return this->template get_val<3, 0>(); }
+};
+
+// block class for the right hand side in state estimation equation
+template <bool sym> using NRSERhs = NRSEUnknown<sym>;
+
+// class of 4*4 (12*12) se gain block
+/*
+[
+   [G, QH]
```

Comment:
```suggestion
   [G, QT]
```

We do the transpose of Q, not hermitian transpose, since we're working with real values here